controllers/controllers_suite_test.go

```diff
@@ -51,9 +51,9 @@ var (
 
 func TestMain(m *testing.M) {
 	setup()
-	defer teardown()
 	code := m.Run()
-	os.Exit(code) //nolint:gocritic
+	teardown()
```

Comment:
Wow. That's an important change

Reply:
Indeed, using `os.Exit` could prevent `defer teardown` from executing. Removing `defer` and making such adjustments allows `teardown` to function properly.